demos/formula_recognition_demo/python/utils.py

```diff
@@ -242,40 +243,37 @@ def _process_decoding_results(self):
             targets = np.argmax(logits, axis=1)
             self.model_status = Model.Status.READY
             return logits, targets
-        self._infer_request_handle_decoder = self._async_infer_decoder(self.row_enc_out,
-                                                                       self.dec_states_c,
-                                                                       self.dec_states_h,
-                                                                       self.output,
-                                                                       self.tgt,
-                                                                       req_id=0
-                                                                       )
+        self._async_infer_decoder(self.row_enc_out,
+                                  self.dec_states_c,
+                                  self.dec_states_h,
+                                  self.output,
+                                  self.tgt.squeeze(axis=0)
+                                  )
 
         return None
 
     def _start_encoder(self, model_input):
-        self._infer_request_handle_encoder = self._async_infer_encoder(model_input, req_id=0)
+        self._async_infer_encoder(model_input)
         self.model_status = Model.Status.ENCODER_INFER
 
     def _start_decoder(self):
-        enc_res = self._infer_request_handle_encoder.output_blobs
-        self._unpack_enc_results(enc_res)
-        self._infer_request_handle_decoder = self._async_infer_decoder(
-            self.row_enc_out, self.dec_states_c, self.dec_states_h, self.output, self.tgt, req_id=0)
+        self._unpack_enc_results()
+        self._async_infer_decoder(self.row_enc_out, self.dec_states_c, self.dec_states_h, self.output, self.tgt)
         self.model_status = Model.Status.DECODER_INFER
 
-    def _unpack_dec_results(self, dec_res):
-        self.dec_states_h = dec_res[self.args.dec_st_h_t_layer].buffer
-        self.dec_states_c = dec_res[self.args.dec_st_c_t_layer].buffer
-        self.output = dec_res[self.args.output_layer].buffer
-        logit = dec_res[self.args.logit_layer].buffer
-        self.logits.append(logit)
+    def _unpack_dec_results(self):
+        self.dec_states_h = self.infer_request_decoder.get_tensor(self.args.dec_st_h_t_layer).data
+        self.dec_states_c = self.infer_request_decoder.get_tensor(self.args.dec_st_c_t_layer).data
+        self.output = self.infer_request_decoder.get_tensor(self.args.output_layer).data
+        logit = self.infer_request_decoder.get_tensor(self.args.logit_layer).data
```

Comment:
For insurance that tensor data will be saved, even you start next request 
```suggestion
        self.dec_states_h = self.infer_request_decoder.get_tensor(self.args.dec_st_h_t_layer).data[:]
        self.dec_states_c = self.infer_request_decoder.get_tensor(self.args.dec_st_c_t_layer).data[:]
        self.output = self.infer_request_decoder.get_tensor(self.args.output_layer).data[:]
        logit = self.infer_request_decoder.get_tensor(self.args.logit_layer).data[:]
```